src/Microsoft.AspNet.FileSystems/Implementation/FileChangeTrigger.cs

```diff
@@ -2,71 +2,56 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Threading;
-using System.Threading.Tasks;
-using System.Text.RegularExpressions;
-using Microsoft.Framework.Expiration.Interfaces;
 
 namespace Microsoft.AspNet.FileSystems
 {
-    internal class FileChangeTrigger : IExpirationTrigger
+    internal class FileChangeTrigger : FileExpirationTriggerBase
     {
-        private CancellationTokenSource TokenSource { get; set; } = new CancellationTokenSource();
+        private static readonly TimeSpan FileCheckInterval = TimeSpan.FromSeconds(3);
+        private readonly IFileSystem _fileSystem;
+        private DateTime _lastRefreshTime;
+        private DateTimeOffset _lastModified;
 
-        public FileChangeTrigger(string pattern)
+        public FileChangeTrigger(IFileSystem fileSystem, string filePath)
         {
-            Pattern = pattern;
+            _fileSystem = fileSystem;
+            FilePath = filePath;
+            UpdateFileInfo();
         }
 
-        public string Pattern { get; private set; }
+        public string FilePath { get; }
 
-        private Regex _searchRegex;
-        private Regex SearchRegex
+        public override bool IsExpired
         {
-            get
-            {
-                if (_searchRegex == null)
-                {
-                    // Perf: Compile this as this may be used multiple times.
-                    _searchRegex = new Regex('^' + Pattern + '$', RegexOptions.IgnoreCase | RegexOptions.ExplicitCapture);
-                }
-
-                return _searchRegex;
-            }
+            get { return base.IsExpired || IsFileChangedInFileSystem(); }
         }
 
-        public bool ActiveExpirationCallbacks
+        // for unit testing.
+        protected virtual DateTime UtcNow
```

Comment:
Consider using the ISystemClock pattern we use elsewhere for testing. See MemoryCache. It limits the number of custom test classes you need to create.